statefun-e2e-tests/statefun-smoke-e2e-driver/src/test/java/org/apache/flink/statefun/e2e/smoke/Utils.java

```diff
@@ -62,12 +62,13 @@ public static SourceCommand aRelayedStateModificationCommand(
 
   /** Blocks the currently executing thread until enough successful verification results supply. */
   static void awaitVerificationSuccess(
-      Supplier<TypedValue> results, final int numberOfFunctionInstances) {
+      Supplier<TypedValue> results, final int numberOfFunctionInstances)
+      throws InvalidProtocolBufferException {
     Set<Integer> successfullyVerified = new HashSet<>();
     while (successfullyVerified.size() != numberOfFunctionInstances) {
       TypedValue typedValue = results.get();
       VerificationResult result =
-          TypedValueUtil.unpackProtobufMessage(typedValue, VerificationResult.parser());
+          VerificationResult.parser().parseFrom(typedValue.getValue().toByteArray());
```

Comment:
I think the `StartedServer` and `SimpleVerificationServer` should really just work directly against `VerificationResult` instead of a `TypedValue` that wraps the result. I don't see any reason that `SimpleVerificationServer` should work against a generic type `T`.

This will also help simplify the Protobuf dependency complication you mentioned in the PR description, since there will be no more `TypedValue` on the surface of these classes.

Reply:
I guess that's for future to write up other kind of e2e tests with different kind of verification? Anyhow that might be far from now.